### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by ryanknapper**
**By submitting this pull request, I confirm the following (please check boxes, eg [X] - no spaces) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

4
---

These edits add a settings section to Advanced DNS settings, allowing the specification of an IP and domain for conditional forwarding.

nuxndfi

settings.php:
	if(isset($setupVars["CONDITIONAL_FORWARDING"])){
		if($setupVars["CONDITIONAL_FORWARDING"] == 1)
		{
			$conditionalForwarding = true;
			$conditionalForwardingDomain = $setupVars["CONDITIONAL_FORWARDING_DOMAIN"];
			$conditionalForwardingIP = $setupVars["CONDITIONAL_FORWARDING_IP"];
		}
		else
		{
			$conditionalForwarding = false;
		}
	} else {
		$conditionalForwarding = false;
	}

savesettings.php:
				// Check if Conditional Forwarding is requested
				if(isset($_POST["conditionalForwarding"]))
				{
					$addressArray = explode(".", $_POST["conditionalForwardingIP"]);
					$reverseAddress = $addressArray[2].".".$addressArray[1].".".$addressArray[0].".in-addr.arpa";
					$extra .= " conditional_forwarding ".$_POST["conditionalForwardingIP"]." ".$_POST["conditionalForwardingDomain"]." $reverseAddress";
				}

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

**Comment by Mcat12** [comment]
Did you update your devel branch with the latest code? You have some changes that have already been made, and your branch conflicts with our branch because of this.

File: php/savesettings.php [hunk 1]
```diff
@@ -119,6 +119,14 @@ function validDomain($domain_name)
 					$extra .= "no-bogus-priv";
 				}
 
+				// Check if Conditional Forwarding is requested
+				if(isset($_POST["conditionalForwarding"]))
+				{
+					$addressArray = explode(".", $_POST["conditionalForwardingIP"]);
+					$reverseAddress = $addressArray[2].".".$addressArray[1].".".$addressArray[0].".in-addr.arpa";
+					$extra .= " conditional_forwarding ".$_POST["conditionalForwardingIP"]." ".$_POST["conditionalForwardingDomain"]." $reverseAddress";
```
Comment (Mcat12) [inline php/savesettings.php]: This is a security issue. Please validate the IP and domain before using it, like how it's done in line 74 and 172:
```php
if (!validIP($primaryIP))
{
    $error .= "Primary IP (".$primaryIP.") is invalid!<br>";
}
// ...
if(!validDomain($domain))
{
    $error .= "Top Domains/Ads entry ".$domain." is invalid!<br>";
}
```

PHP makes it easy to create security issues, so always remember to never trust the client's data. In the current code, someone could add a script in the domain or IP address field and have it executed.

File: settings.php [hunk 2]
```diff
@@ -257,6 +471,51 @@
...
+			$conditionalForwardingDomain = $setupVars["CONDITIONAL_FORWARDING_DOMAIN"];
+			$conditionalForwardingIP = $setupVars["CONDITIONAL_FORWARDING_IP"];
+		}
+		else
+		{
+			$conditionalForwarding = false;
+		}
+	} else {
+		$conditionalForwarding = false;
+	}
```
Comment (Mcat12) [inline settings.php]: You could combine the first two if statements:
```php
if(isset($setupVars["CONDITIONAL_FORWARDING"]) && $setupVars["CONDITIONAL_FORWARDING"] == 1) {
	$conditionalForwarding = true;
	$conditionalForwardingDomain = $setupVars["CONDITIONAL_FORWARDING_DOMAIN"];
	$conditionalForwardingIP = $setupVars["CONDITIONAL_FORWARDING_IP"];
} else {
	$conditionalForwarding = false;
}
```

**Comment by ryanknapper** [comment]
Good notes, thanks.

**Pull request closed without merging**